### ISSUE TITLE
Feature Porting

### ISSUE THREAD

**Issue by roboticlemon** [Flova/ros_gremsy#4]
Hi @Flova,

I have an organisation maintained internal code base for controlling Gremsy Gimbals and have a properly done Pixy U URDF with configuration settable prismatic joints and joint state revolute joints. I am looking to make it open source for the community.

![image](https://user-images.githubusercontent.com/2769347/96461702-6e3c1300-1270-11eb-926f-4f92e384a39c.png)

After polishing up my action interface and making it generic I'd love to add more functionality to this repo instead of having 2 separate repos. At the moment my interface operates as an action service instead of just a topic so there might be some work for me to support both in your existing code. I also modified the gremsySDK to be better object oriented with higher level abstractions but this is probably a bad idea with the changing sdk code from Gremsy.

I'll take a look through your repo more thoroughly and see what I need to change to start mergeing changes!

Let me know your thoughts,

Cheers

**Comment by Flova** [comment]
Hi @ingramator,

this sounds very good. Thank you very much!

We definitely look forward to this!

Cheers

**Comment by Flova** [comment]
Any updates on this @ingramator ? : )

**Comment by roboticlemon** [comment]
@Flova I completely forgot about this! My apologies, I'll try and get some stuff up this week.

**Comment by Flova** [comment]
Nice, it's fine, no hurry.

**Comment by roboticlemon** [comment]
@Flova How I have my internal repo setup I have separate packages for separate concerns. Most ros robots are setup like this and it might be a good idea to do it for this repo as we add functionality. For example for my robot_description ive made it would be good to have a gremsy_description package that just has urdf related code in it. We could also have a gremsy_msgs package for custom messages in this workspace so when you need to interface with them from another package you only need the message definitions and not the rest of the codebase.

Let me know your thoughts!

**Comment by Flova** [comment]
I totally agree with you. I also structure my other ROS projects like this. Until now this was "small" enough to skip this but if we add a propper urdf description and so on it would be indeed useful.

**Comment by roboticlemon** [comment]
Excellent, if you want to refactor your current code into it's own package (maybe like gremsy_base?) then it will be easier for me to send a PR with a draft decription package. We can keep all gremsy model descriptions in there if we get community contributions.

**Comment by Flova** [comment]
Sounds nice. I am a bit busy this week but would do it in the near future.

**Comment by Flova** [comment]
@ingramator Sorry, that it took me so long to complete the restructuring. But I finally finished it and I hope it is something we can work with.
